### ISSUE TITLE
Azure: SignIn hangs forever

### ISSUE THREAD

**Issue by andreas-schneider** [microsoft/vscode-azure-account#179]
Hi everyone!

I'm on Linux and not behind a proxy.

If I select "Azure: SignIn" it starts a new tab in Firefox on an URL with some nonce
http://localhost:42247/signin?nonce=......
and it tries to load a page forever, nothing happens. 
Back in VS Code it says I'm not signed in.

Version: 1.45.0-insider
Commit: 6e9a34c0de4181a99f11e106733ee3e8c9db79f1
Date: 2020-04-16T05:44:19.773Z
Electron: 7.2.1
Azure Account Version: 0.8.9 but I have the same issue with other versions.

Also, the same issue on the stable version of VS Code.

On Windows with Firefox it works.

Has anyone an idea what I can do? Thanks!

**Comment by poiu2131** [comment]
@andreas-schneider Hi Andress, I have the same issue as your. Did you already fund the solution to solve it? Great thanks.

**Comment by jungaretti** [comment]
I'm having the same issue both when I use the `Azure: Sign In` command and when I click on a sign-in button in an Azure extension's viewlet. My browser launches and hangs on `http://localhost:38309/signin?nonce=V9Gb5t4q7pL+aZgxmUbH+Q==`. The sign-in page never loads, but if I ditch the `/signin` request and visit `localhost:38309`, the "you are signed in now and can close this page" page appears right away.

#### Steps to reproduce with command

1. Disable all extension except for Azure Account
2. Use the `Azure: Sign In` command
3. Browser opens, but the `localhost` server never loads

#### Steps to reproduce with sign-in button:

1. Disable all extension except for Azure Account and Azure Storage
2. Open the Azure viewlet and click on Azure Storage's "Sign in to Azure..." button
3. Browser opens, but the `localhost` server never loads

#### About output

Version: 1.49.1
Commit: 58bb7b2331731bf72587010e943852e13e6fd3cf
Date: 2020-09-16T23:23:33.049Z
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Linux x64 5.8.10-arch1-1

**Comment by jungaretti** [comment]
You can use the `Azure: Sign In with Device Code` command as a workaround!

**Comment by phil-w** [comment]
Yes, it says I should try the "device code" thing, but what is a "device code", and where would I get one? What does that mean?

**Comment by jungaretti** [comment]
It’s another command in VS Code. It’ll prompt you with a page to visit from your browser where you can sign in, generate a code, and paste it into VS Code (versus running some sort of local server and only answering the first request). Give it a try! I included the exact command in an earlier comment

**Comment by phil-w** [comment]
Thanks - it was not visible the first few times I tried this, then eventually it came up and worked perfectly.

It appears in a little blue box bottom right of the screen and there's a URL plus a code. 
You just have to click it and it does whatever it is that it wants to do and doesn't get in the way any more.

I don't know why it didn't come up for me like that straight away.
Anyway, if it's not there, just keep rebooting or closing/ opening and eventually it will show up.

**Comment by wwlorey** [comment]
It would be helpful to recommend the "Sign-in with Device Code" command after a timeout error with the standard "Sign-in" command

**Comment by TimMarshAU** [comment]
I'm on Always on VPN to connect to work - Win10. I changed proxy setting in Preferences (search for "Proxy Support") from "override" to "off". "On" also works.

**Comment by mccullj** [comment]
In case this helps devs or anyone reading this ticket:

I was running into issues related to my corporate proxy attempting to sign in via extension ver. 0.10.1. Tried many combinations of proxy settings (on/override/off, redefining ENV proxies in VS Code, etc.) per this and other related issues, regular sign-in and device code sign-in, no luck. What did work, in my case, was downgrading the extension to 0.9.8 and then using device code sign-in (works without additional proxy settings in VS Code, using ENV proxy settings).

**Comment by TimMarshAU** [comment]
I finally got this working by changing our Bluecoat proxy to enable tunnelling to the relevant URLs. Bluecoat was intercepting the SSL connections, trying to inspect traffic, which SSL doesn't like. After this change, VS Code was no longer using device code logins (which was causing the callback to VS Code to fail, so the Azure subscriptions never retrieved, so it hung on logging in stage), and the native localhost browser authentication worked.

Enable tunnelling on (at least, there may be more) these
https://management.azure.com/*
https://functionscdn.azureedge.net/*
https://vscode-redirect.azurewebsites.net/*
https://login.microsoftonline.com/*

**Comment by alexweininger** [comment]
Hi all, we've created two development builds with proposed fixes for proxy issues. If you are still facing issues with the extension while using a proxy it would be a huge help if you could test these. #714 contains details and instructions on how to do so.

**Comment by C0smin** [comment]
Is `Sign-in with Device Code` available anymore? I cannot find it in the latest version of the tools
